Support for both Brightway 2 and 2.5

cmutel opened the pull request:
🔨 Work in progress 🔨

Factors out Brightway functionality, and imports the correct function depending on which version is installed.

* To install the 2.5 version via pip: `pip install premise[bw25]`.
* On `conda` we might be able to do the same variant, or just create two different packages.

To support this, I moved from `setup.py` to `pyproject.toml`. This also gives us:

* The new standard packaging pipeline 
* Automatic building of wheels (run `python -m build`)
* One place to define the version: `__init__.py`
* One place to define the requirements: `requirements.txt`
* Inclusion of pytest and other config data (`pytest.ini` is removed)

Note that 2.5 support requires `bw2data 4.0.DEV32`, which isn't yet released (will be in ~1 hour from this comment).

We need some testing before this should be released. I have tested the 2.5 functionality, and it builds a matrix and gives a sensible result. However, it would be nice to have a known good result, and test both BW2 and BW2.5 scores. @romainsacchi Could you give a known good result please?

# TODO
- [x] Add notification for `bw2data` version imported, and bw2/2.5
- [x] Test export to normal DB on BW2
- [x] Test export to normal DB on BW2.5
- [ ] Check packaging still works on CI

cmutel committed:
Add some basic work on 2.5 compat

cmutel committed:
Merge branch 'master' into bw25compat

cmutel committed:
Separate out Brightway functionality

cmutel committed:
Experimental support for both bw2 and bw25

romainsacchi committed:
Black reformating

tngTUDOR commented on premise/ecoinvent_modification.py:
```diff
@@ -49,9 +49,16 @@
+    import bw_processing
+
+    from .brightway25 import write_brightway_database
+except ImportError:
```
suggestion (non-blocking): It would be useful to warn the user if the import goes OK or (bw2) or wrong.

cmutel replied:
Added to `todo`

cmutel committed:
Remove 3.11 restriction

cmutel committed:
Add log message on BW 2 or 2.5

cmutel committed:
Use docs section in CI config

cmutel committed:
Don't install on 3.12

The dependency `sparse` is broken, it imports `Iterable` from collections which doesn't work.

cmutel committed:
Merge branch 'bw25compat' of https://github.com/cmutel/rmnd-lca into bw25compat

cmutel committed:
Merge remote-tracking branch 'upstream/master' into bw25compat

romainsacchi committed:
Black reformating

cmutel committed:
Merge remote-tracking branch 'upstream/master' into bw25compat
